server/src/pages/homepage/view.js

```diff
@@ -16,6 +17,7 @@ class Head extends React.Component {
       <reactruntime.HeadTemplate {...this.props}>
         <link rel="stylesheet" href={ this.props.staticLink("/static/css/home.css") } />
         <script src={this.props.staticLink("/static/js/UITour-lib.js")} async></script>
+        <script src={ this.props.staticLink("/static/js/wantsauth.js") } />
```

Comment:
There's a bug with the way auth currently works: the pages send a request to `/api/login` every time.  Once the client has the cookies, pages should not send login requests.  It looks like that's what My Shots meant to do with its conditional inclusion of `wantsauth`, but it's looking at a undefined property in the condition.

Let's do this:

- If there's a device id, then don't include `wantsauth`.
- If there's a device id but not account id, and the page wants to know if the user has FxA auth, then include `wantsauth`.

That means we need to add something to indicate that the page wants FxA auth after a successful FxA login.  A query param should work.

Reply:
@chenba since we are redirecting to /shots on successful login, we can safely remove wantsauth.js from homepage view and avoid /api/login calls

Reply:
We did discuss redirecting users back to wherever they init'd the login so that includes the homepage.  `/shots` was the fallback when there's no redirect uri.  Is that not the case?

And all the pages need to be fixed.  I was thinking doing that in this PR since we are knee deep in auth.  But I could file another bug if that's better.

Reply:
@chenba As discussed, will fix auth bug in this PR itself by moving conditional wantsauth.js include at one place in reactrender.js

Will update homepage to redirect user back to homepage after successful FxA login

Reply:
hmm actually `reactruntime` would be more appropriate.

Reply:
@chenba Updated PR to fix auth bug and load wantsAuth.js inside reactruntime only when needed